cmd/gen-go-sample/testdata/sample_enum.want

```diff
@@ -0,0 +1,62 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Code generated by protoc-gen-go_gapic. DO NOT EDIT.
+
+package main
+import(
+ "context"
+ "flag"
+ "fmt"
+ "log"
+
+foo "path.to/client/foo"
+foopb "path.to/pb/foo"
+)
+// [START awesome_region]
+
+func sampleUnaryMethod(myFruit foopb.AType_FruitEnum) error {
+	ctx := context.Background()
+	c, err := foo.NewClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	// myFruit := foopb.AType_APPLE
+	req := &foopb.InputType{
+		E: myFruit,
+	}
+	resp, err := c.UnaryMethod(ctx, req)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("call finished.\n")
+	return nil
+}
+
+// [END awesome_region]
+
+func main() {
+	myFruit := flag.String("my_fruit", "APPLE", "")
+	flag.Parse()
+	myFruit2, ok := foopb.AType_FruitEnum_name[*myFruit]
```

Comment:
IIUC, the name `myFruit2` is autogenerated from `myFruit`? Having the suffix 2 does not read very friendly to me.
I would suggest either:
 
(a) using `myFruitArg`
```
myFruitArg := flag.String("my_fruit", ...)
...
myFruit, ok := foopb.AType_FruitEnum_name[*myFruitArg]
```
OR 

(b) using `myFruitEnum` (or similar)
```
myFruit := flag.String("my_fruit", ...)
...
myFruitEnum, ok := foopb.AType_FruitEnum_name[*myFruit]
```

WDYT? I personally prefer (a). It's fine to do this in a follow-up PR, but I think we should.

Reply:
Sounds good. I'll make `myFruitArg` the base name, and if that's already taken we use a numeric suffix to disambiguate.

Reply:
Done